### PR TITLE
namespaced-openvpn: init at 0.6.0

### PR DESCRIPTION
**Pull request by lodi**
## Description of changes

`namespaced-openvpn` is a python script to automate the setup of isolated OpenVPN tunnels using Linux Network Namespaces.  This allows secure per-application control of what data travels over the VPN while leaving the rest of the system unaffected.

https://github.com/slingamn/namespaced-openvpn

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/namespaced-openvpn/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  postPatch = ''
+    substituteInPlace namespaced-openvpn \
+      --replace-fail "/usr/sbin/openvpn" "${openvpn}/bin/openvpn"
+    substituteInPlace namespaced-openvpn \
+      --replace-fail "/sbin/ip" "${iproute2}/bin/ip"
+    substituteInPlace namespaced-openvpn \
+      --replace-fail "/usr/bin/nsenter" "${util-linux}/bin/nsenter"
+    substituteInPlace namespaced-openvpn \
+      --replace-fail "/bin/mount" "${util-linux}/bin/mount"
+    substituteInPlace namespaced-openvpn \
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/namespaced-openvpn/default.nix:34]:
```suggestion
      --replace-fail "/usr/sbin/openvpn" "${openvpn}/bin/openvpn" \
      --replace-fail "/sbin/ip" "${iproute2}/bin/ip" \
      --replace-fail "/usr/bin/nsenter" "${util-linux}/bin/nsenter" \
      --replace-fail "/bin/mount" "${util-linux}/bin/mount" \
```

File: pkgs/tools/networking/namespaced-openvpn/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{ lib,
+  fetchFromGitHub,
+  buildPythonPackage,
+  python3Packages,
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/namespaced-openvpn/default.nix:4]:
```suggestion
```
that's unused

**Comment by lodi** [comment]
@SuperSandro2000 Great, thank you for the tips.  One other thing I wasn't sure of: this project inherently depends on Linux functionality.  Is there something I need to do to hide or restrict this from building on other platforms?

**Comment by SuperSandro2000** [comment]
You already set meta.platforms, there is not other way to do that.

**Pull request merged**